### PR TITLE
feat: Offset tracker based on SlaOS API

### PR DESCRIPTION
**Pull request by nimbol**

### CODE REVIEW

File: src/clients/slaos.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        params = {
+            "key": datastream_key,
+            "limit": "1",
+        }
+        headers = {
+            "Content-Type": "application/json",
+        }
+        # TODO: handle all manner of exceptions in this block
+        response = self.client.get(url, params=params, headers=headers)
+        response.raise_for_status()
```
Comment (nimbol) [inline src/clients/slaos.py:32]: @samtin0x Will there be a need to handle this more elegantly? We may get errors from misconfigurations or even from the API being unavailable. Are these errors okay or do we fall back to the start time that's in the config?

File: src/clients/slaos.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from datetime import datetime
+import httpx
+
+from src.config.models.offset import SLIIndexedDocument
+from src.config.models.output import RatedOutputConfig
```
Comment (nimbol) [inline src/clients/slaos.py:5]: I've used another area's config object here. There is 100% overlap in config. I should probably restructure the config yaml instead.

File: src/indexers/offset_tracker/rated.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    def initialise_offset(self) -> None:
+        if self.config.override_start_from:
+            offset = self.config.start_from
+        else:
+            api_timestamp = self.get_offset_from_api()
+            if api_timestamp is not None:
+                from_epoch = api_timestamp - datetime(1970, 1, 1, tzinfo=UTC)
+                offset = int(from_epoch.total_seconds())
+            else:
+                offset = self.config.start_from
```
Comment (nimbol) [inline src/indexers/offset_tracker/rated.py:36]: These offsets are different. One is seconds from epoch, the other is milliseconds.

File: src/clients/slaos.py (hunk 1 above)
Comment (nimbol) [reply src/clients/slaos.py:32]: > I'd suggest if there's another option where we only fetch it once at the beginning. So we fetch it and don't make more network requests and then if instance breaks or something, on restart we fetch it from the API again

Yes, that's how this is intended to work. That doesn't mean there will never be errors.

File: src/clients/slaos.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+from datetime import datetime
+import httpx
+import stamina
+from pydantic import BaseModel
+
+from src.config.models.output import RatedOutputConfig
+
+
+class SlaosClient:
+    def __init__(self, config: RatedOutputConfig):
```
Comment (dswistowski) [inline src/clients/slaos.py:10]: not sure if it should be a class - it does have one method and no state -  creating two functions: `get_latest_ingest_timestamp(ingest_url, datastream_key)` and `make_ingest_url(*, base_url, id, key)` might be just simpler

**Pull request merged**